src/core/print.ml

```diff
@@ -300,26 +300,27 @@ let pp_ctxt : ctxt pp = fun ppf ctx ->
       let pp_def ppf t = out ppf " ≔ %a" pp_term t in
       let pp_decl ppf (x,a,t) =
         out ppf "%a%a%a" pp_var x pp_type a (Option.pp pp_def) t in
-      List.pp pp_decl "\n, " ppf (List.rev ctx);
-      if ctx <> [] then out ppf " ";
-      out ppf "\n⊢ "
+
+      out ppf "%a%s⊢ "
+        (List.pp pp_decl ",@ ") (List.rev ctx)
+        (if ctx <> [] then "@ " else "")
     end
 
 let pp_typing : constr pp = fun ppf (ctx, t, u) ->
-  out ppf "%a%a : %a" pp_ctxt ctx pp_term t pp_term u
+  out ppf "%a%a@ : %a" pp_ctxt ctx pp_term t pp_term u
 
 let pp_constr : constr pp = fun ppf (ctx, t, u) ->
-  out ppf "%a%a ≡ %a" pp_ctxt ctx pp_term t pp_term u
+  out ppf "%a%a@ ≡ %a" pp_ctxt ctx pp_term t pp_term u
 
-let pp_constrs : constr list pp = fun ppf ->
-  List.iter (out ppf "\n  ; %a" pp_constr)
+let pp_constrs : constr list pp =
+  List.pp pp_constr ";@ "
 
 (* for debug only *)
-let pp_metaset : MetaSet.t pp = fun ppf ->
-  MetaSet.iter (out ppf "%a," pp_meta)
+let pp_metaset : MetaSet.t pp =
+  D.iter ~sep:(fun fmt () -> Format.fprintf fmt ",@ ") MetaSet.iter pp_meta
```

Comment:
You could perhaps add in base.ml a function for this separator.
But shouldn't write ", @" (break after space)? Otherwise, don't you get a space at the beginning of the next line?

Reply:
That's not how it works: `@` is only a prefix, `@ ` is either a space or a newline.
Similarly, `@,` is either no space or a newline, and `@;` is a custom break hint.

Reply:
Oh  I see. Thank you for the explanation.